lib/active_fedora/solr_hit.rb

```diff
@@ -0,0 +1,71 @@
+module ActiveFedora
+  class SolrHit < Delegator
+    def self.for(hit)
+      return hit if hit.is_a? ActiveFedora::SolrHit
+
+      SolrHit.new(hit)
+    end
+
+    delegate :models, to: :classifier
+
+    def __getobj__
+      @document # return object we are delegating to, required
+    end
+
+    alias static_config __getobj__
+
+    def __setobj__(obj)
+      @document = obj
+    end
+
+    attr_reader :document
+
+    def initialize(document)
+      super
+      @document = document
+    end
+
+    def id
+      document[ActiveFedora.id_field]
```

Comment:
I went back and forth about whether this should be `document` or `self`. I landed on `document`, because I anticipate (some day) discouraging consumers from trying to access the raw solr response (and strongly suggesting `instantiate_with_json` or `reify` instead). I addressed all non-test hash access in ActiveFedora, so I'm optimistic that this may be possible.